### DISCUSSION TITLE
Create launchd config for MacOS for use with `watch` option

### DISCUSSION THREAD

**Discussion by Bas-Man** [Taxel/PlexTraktSync#889] [Show and tell]
Requirements
==========

1. Directory named `LaunchAgents` under personal `Library` directory

Steps
------
 - Under the directory `LaunchAgents` create file called something like `net.PlexTraktSync.LaunchAtLogin.plist`
   - This is part of the naming conventions
 - Place the following contents into the file replacing `COMMAND` with the full path to the `plextraktsync` command. This can be found using the `which` command in the terminal.

```text
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>Label</key>
	<string>net.PlexTraktSync.LaunchAtLogin</string>
	<key>ProgramArguments</key>
	<array>
		<string>COMMAND</string>
		<string>watch</string>
	</array>
	<key>LimitLoadToSessionType</key>
	<string>Aqua</string>
	<key>RunAtLoad</key>
	<true/>
	<key>ExitTimeOut</key>
	<integer>0</integer>
	<key>ProcessType</key>
	<string>Background</string>
</dict>
</plist>
```

- load the newly created launchd config using `launchctl load Library/LaunchAgents/net.PlexTraktSync.LaunchAtLogin.plist`
  - This will register the config with launchd. `plextraktsync` should now run any time the machine runs with the user logged in. For my plex server I always have it boot up into the account that runs Plex.

You can also disable this by using `launchctl unload Library/LaunchAgents/net.PlexTraktSync.LaunchAtLogin.plist`

-- Note that the `launchctl` command is being run from the home directory of the user for which both `Plex` and `PlexTraktSync` have been installed.

**Comment by glensc** [comment]
Attempt to add a sub-command for this:
- https://github.com/Taxel/PlexTraktSync/pull/890

please command on the questions there

**Comment by glensc** [comment]
How to check output or whether it's working? or what the status 78 means?

```
$ launchctl list |head
PID	Status	Label
-	78	com.github.plextraktsync.watch
```

**Reply by glensc** [reply]
There's some log in `/var/log/system.log`:

```
Oct 20 19:34:36 vinagre com.apple.xpc.launchd[1] (com.github.plextraktsync.watch[45029]): Service could not initialize: 19H2026: xpcproxy + 14537 [839][FB7D36FA-1F35-3A7A-B0FF-B1FAA70E4ED5]: 0x2
Oct 20 19:34:36 vinagre com.apple.xpc.launchd[1] (com.github.plextraktsync.watch[45029]): Service exited with abnormal code: 78
Oct 20 19:34:37 vinagre com.apple.xpc.launchd[1] (com.apple.MRTa): Service only ran for 0 seconds. Pushing respawn out by 10 seconds.
```

doc:
- https://stackoverflow.com/a/56188963/2314626
